lightning/src/util/events.rs

```diff
@@ -81,6 +82,39 @@ impl_writeable_tlv_based_enum!(PaymentPurpose,
 	(2, SpontaneousPayment)
 );
 
+/// When the payment failure took place and extra details about it. [`Failure::OnPath`] may contain
```

Comment:
This should say "payment path failure", I believe. Or maybe "HTLC failure" to align with more specific naming?